packages/salesforcedx-vscode-core/src/commands/forceSourcePull.ts

```diff
@@ -47,9 +68,9 @@ export class ForceSourcePullExecutor extends SfdxCommandletExecutor<{}> {
 const workspaceChecker = new SfdxWorkspaceChecker();
 const parameterGatherer = new EmptyParametersGatherer();
 
-export async function forceSourcePull(this: FlagParameter<string>) {
```

Comment:
@gbockus-sf Yes I do see the `this` param used in other places for the cmd funcs. 
Guess it has to do with how they are passed to `registerCommand`: `export function registerCommand(command: string, callback: (...args: any[]) => any, thisArg?: any): Disposable;` where `@param thisArg The `this` context used when invoking the handler function.`

But all our tests are passing even with this change. 
@jag-sfdc Is the `this` convention required? Should we keep it?

Reply:
I don't see a reason to keep this convention (unless I'm missing something). Agree with an earlier comment that it is confusing.

Reply:
Actually when testing manually found that without the `this` the flags are not passed when the command is invoked. Guess it is a vscode specific thing. But neither our unit or integ tests were able to catch this. Going to check if existing tests can be updated to be able to catch this.

Reply:
So I did some digging b/c who likes mysteries.  The best doc I found around it was this: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#specifying-the-type-of-this-for-functions.  So basically it's a way to tell Typescript, `I expected this function to be running in this 'this' context when executing`.  Super weird.  In the generated js there is no parameter. It's used only for typechecking. Also the vscode command for registers is any'd to the max so I'm wondering if it even buys us anything to type this parameter.